### PR TITLE
python3Packages.rencode: switch to fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes
switch to `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 163576` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>ib-controller</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>deluge</li>
    <li>python310Packages.rencode</li>
    <li>python39Packages.rencode</li>
    <li>run-scaled</li>
    <li>xpra</li>
    <li>zeronet</li>
  </ul>
</details>

File: pkgs/development/python-modules/rencode/default.nix [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
-  src = fetchgit {
-    url = "https://github.com/aresch/rencode";
+  src = fetchFromGitHub {
+    owner = "aresch";
+    repo = "rencode";
     rev = "b45e04abdca0dea36e383a8199783269f186c99e";
-    sha256 = "b4bd82852d4220e8a9493d3cfaecbc57b1325708a2d48c0f8acf262edb10dc40";
+    sha256 = "sha256-tL2ChS1CIOipST08+uy8V7EyVwii1IwPis8mLtsQ3EA=";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rencode/default.nix:16]:
```suggestion
```
The repo has no submodules.

File: pkgs/development/python-modules/rencode/default.nix [hunk 2]
```diff
@@ -1,17 +1,18 @@
 { lib
 , buildPythonPackage
-, fetchgit
+, fetchFromGitHub
 , cython
 }:
 
 buildPythonPackage {
   pname = "rencode";
   version = "git20150810";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rencode/default.nix:9]:
```suggestion
  version = "1.0.6";
```

File: pkgs/development/python-modules/rencode/default.nix [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 
-  src = fetchgit {
-    url = "https://github.com/aresch/rencode";
+  src = fetchFromGitHub {
+    owner = "aresch";
+    repo = "rencode";
     rev = "b45e04abdca0dea36e383a8199783269f186c99e";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rencode/default.nix:14]:
```suggestion
    rev = "v${version}";
```

File: pkgs/development/python-modules/rencode/default.nix [hunk 4]
```diff
@@ -1,17 +1,18 @@
 { lib
 , buildPythonPackage
-, fetchgit
+, fetchFromGitHub
 , cython
 }:
 
 buildPythonPackage {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/rencode/default.nix:7]:
```suggestion
buildPythonPackage rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 163576` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>ib-controller</li>
    <li>xpraWithNvenc</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>deluge</li>
    <li>python310Packages.rencode</li>
    <li>python39Packages.rencode</li>
    <li>run-scaled</li>
    <li>xpra</li>
    <li>zeronet</li>
  </ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/163576

1 package failed to build:
ib-controller

7 packages built:
deluge python310Packages.rencode python39Packages.rencode run-scaled xpra xpraWithNvenc zeronet
```

**Pull request merged**